src/ucp/core/ucp_am.c

```diff
@@ -458,17 +471,55 @@ ucp_am_send_short(ucp_ep_h ep, uint16_t id, uint16_t flags, const void *header,
      */
     ucs_assert((length == 0ul) || (header_length == 0));
     ucs_assert(!(flags & UCP_AM_SEND_REPLY));
-    UCS_STATIC_ASSERT(sizeof(ucp_am_hdr_t) == sizeof(uint64_t));
-    hdr.am_id         = id;
-    hdr.flags         = flags;
-    hdr.header_length = header_length;
+    ucp_am_fill_short_header(&hdr, id, flags, header_length);
 
     sbuf = (header_length != 0) ? (void*)header : (void*)payload;
 
     return uct_ep_am_short(am_ep, UCP_AM_ID_SINGLE, hdr.u64, sbuf,
                            length + header_length);
 }
 
+static UCS_F_ALWAYS_INLINE ucs_status_t
+ucp_am_send_short_reply(ucp_ep_h ep, uint16_t id, uint16_t flags,
+                        const void *header, size_t header_length,
+                        const void *payload, size_t length)
+{
+    uct_ep_h am_ep   = ucp_ep_get_am_uct_ep(ep);
+    size_t tx_length;
+    ucp_am_hdr_t hdr;
+    const void *data;
+    void *tx_buffer;
+    ucs_status_t status;
+
+    ucs_assert(flags & UCP_AM_SEND_REPLY);
+    ucs_assert((length == 0ul) || (header_length == 0));
```

Comment:
it is safe to compare with `0`

Reply:
yes, but will fix it for the consistency